gd32e10x/cmsis/gd/gd32e10x/include/gd32e10x.h

```diff
@@ -50,10 +50,10 @@ OF SUCH DAMAGE.
 
 /* define value of high speed crystal oscillator (HXTAL) in Hz */
 #if !defined  HXTAL_VALUE
-  #ifdef GD32E103V_EVAL
+  #ifdef CONFIG_BOARD_GD32E103V_EVAL
   #define HXTAL_VALUE    ((uint32_t)8000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_8M  HXTAL_VALUE
-  #elif defined(GD32E103R_START)
+  #elif defined(CONFIG_BOARD_GD32E103R_START)
```

Comment:
HAL headers can't depend on Kconfig variables. Instead, Zephyr build system should insert the `HXTAL_VALUE` based on a Kconfig or DT value. See https://github.com/zephyrproject-rtos/zephyr/commit/a93908563631ddba6308587e77cffd70bde30411

Reply:
I tried to use HXTAL_VALUE, but I'm a bit unsure if it's the same as your idea.